src/foam/dao/JDAO.java

```diff
@@ -7,24 +7,117 @@
 package foam.dao;
 
 import foam.core.ClassInfo;
-import foam.lib.json.Outputter;
-import foam.lib.json.OutputterMode;
+import foam.core.FObject;
+import foam.core.X;
+import foam.mlang.order.Comparator;
+import foam.mlang.predicate.Predicate;
+import foam.nanos.auth.User;
+import foam.util.SafetyUtil;
+
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.TimeZone;
 
 public class JDAO
-  extends AbstractJDAO
+  extends ProxyDAO
```

Comment:
Why are we not modelling this in JS and generating this class rather than writing it by hand?

Reply:
We should be. I’ll make it so.

Reply:
Finished modelling JDAO